logsmanagement/logsmanagement.c

```diff
@@ -0,0 +1,637 @@
+#include "../daemon/common.h"
+#include "../libnetdata/libnetdata.h"
+#include <assert.h>
+#include <inttypes.h>
+#include <lz4.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <time.h>
+#include <uv.h>
+#include "circular_buffer.h"
+#include "logsmanagement_conf.h"
+#include "db_api.h"
+#include "file_info.h"
+#include "helper.h"
+#include "query.h"
+#if LOGS_MANAGEMENT_STRESS_TEST
+#include "query_test.h"
+#endif  // LOGS_MANAGEMENT_STRESS_TEST
+#include "parser.h"
+#include "flb_plugin.h"
+#include "tail_plugin.h"
+
+static struct config log_management_config = {
+    .first_section = NULL,
+    .last_section = NULL,
+    .mutex = NETDATA_MUTEX_INITIALIZER,
+    .index = {
+            .avl_tree = {
+                    .root = NULL,
+                    .compar = appconfig_section_compare
+            },
+            .rwlock = AVL_LOCK_INITIALIZER
+    }
+};
+
+struct File_infos_arr *p_file_infos_arr = NULL;
+uv_loop_t *main_loop; 
+
+volatile sig_atomic_t p_file_infos_arr_ready = 0;
+int g_logs_manag_update_every = 1;
+
+/**
+ * @brief Initialise monitoring of a log source.
+ * @param[in] filename The filename (if the log source is a file) of the log source.
+ * @param[in] log_type The type of the log source (web log, system logs etc.)
+ * @param[in] circular_buffer_initial_size Initial size of the buffer dedicated to
+ * this log source.
+ * @param[in] circular_buffer_max_size Maximum memory the buffer dedicated to this
+ * log source can occupy.
+ * @param[in] compression_accel Compression acceleration factor for the lz4 
+ * compression step (see following link for more information)
+ * https://github.com/lz4/lz4/blob/90d68e37093d815e7ea06b0ee3c168cccffc84b8/lib/lz4.h#L195)
+ * @param[in] buff_flush_to_db_interval Interval of how often to write compressed
+ * logs to disk.
+ * @param[in] blob_max_size Maximum disk space that compressed logs can occupy 
+ * before the old ones get overwritten. 
+ * @return Pointer to initialised struct File_info p_file_info associated to
+ * this log source, or NULL if any errors are encountered.
+ */
+static struct File_info *monitor_log_file_init(const char *filename, 
+                                               const enum log_source_t log_type,
+                                               const size_t circular_buffer_max_size, 
+                                               const int compression_accel, 
+                                               const int buff_flush_to_db_interval,
+                                               const int64_t blob_max_size,
+                                               const int update_every) {
+    int rc = 0;
+
+    info("Initializing log source collection for %s", filename);
+
+    struct File_info *p_file_info = callocz(1, sizeof(struct File_info));
+
+    p_file_info->filename = (char *) filename;            
+    p_file_info->file_basename = get_basename(filename); // NOTE: get_basename uses strdupz. freez() if necessary!
```

Comment:
I did not understand the `freez() if necessary`. The function `strdupz`works like `strdup`, so like `strdup` we need to call `freez`.

Reply:
Yes, I am doing that. It's just a note for any future developer, to not forget to `free` the `p_file_info->file_basename` variable, if used in a different way.